core/src/main/java/io/grpc/internal/KeepAliveManager.java

```diff
@@ -234,18 +233,57 @@ public synchronized void onTransportShutdown() {
     }
   }
 
-  private class KeepAlivePingCallback implements ClientTransport.PingCallback {
+  public interface KeepAlivePinger {
+    /**
+     * Sends out a keep-alive ping.
+     */
+    void ping();
+
+    /**
+     * Callback when Ping Ack was not received in KEEPALIVE_TIMEOUT. Should shutdown the transport.
+     */
+    void onPingTimeout();
+
+    /**
+     * Specifies whether the pinger should keep sending keep-alive pings when the transport has no
+     * active RPCs.
+     */
+    boolean stillPingWhileIdle();
```

Comment:
Why is this a method instead of staying as a field? This must always return the same value, otherwise the code may break.

I don't understand what is special about this one configuration option such that it would be added to the Pinger.